src/ObjLoading/Game/T6/AssetLoaders/AssetLoaderRawFile.cpp

```diff
@@ -20,13 +26,61 @@ bool AssetLoaderRawFile::CanLoadFromRaw() const
     return true;
 }
 
-bool AssetLoaderRawFile::LoadFromRaw(
-    const std::string& assetName, ISearchPath* searchPath, MemoryManager* memory, IAssetLoadingManager* manager, Zone* zone) const
+bool AssetLoaderRawFile::LoadAnimtree(
+    const SearchPathOpenFile& file, const std::string& assetName, ISearchPath* searchPath, MemoryManager* memory, IAssetLoadingManager* manager)
 {
-    const auto file = searchPath->Open(assetName);
-    if (!file.IsOpen())
+    const auto uncompressedBuffer = std::make_unique<char[]>(static_cast<size_t>(file.m_length + 1));
+    file.m_stream->read(uncompressedBuffer.get(), file.m_length);
+    if (file.m_stream->gcount() != file.m_length)
+        return false;
+    uncompressedBuffer[static_cast<size_t>(file.m_length)] = '\0';
+
+    const auto compressionBufferSize = static_cast<size_t>(file.m_length + 1 + sizeof(uint32_t) + COMPRESSED_BUFFER_SIZE_PADDING);
+    auto* compressedBuffer = static_cast<char*>(memory->Alloc(compressionBufferSize));
+
+    z_stream_s zs{};
+
+    zs.zalloc = Z_NULL;
+    zs.zfree = Z_NULL;
+    zs.opaque = Z_NULL;
+    zs.avail_in = static_cast<uInt>(file.m_length + 1);
+    zs.avail_out = compressionBufferSize;
+    zs.next_in = reinterpret_cast<const Bytef*>(uncompressedBuffer.get());
+    zs.next_out = reinterpret_cast<Bytef*>(&compressedBuffer[sizeof(uint32_t)]);
+
+    int ret = deflateInit2(&zs, Z_DEFAULT_COMPRESSION, Z_DEFLATED, -DEF_WBITS, DEF_MEM_LEVEL, Z_DEFAULT_STRATEGY);
+
+    if (ret != Z_OK)
+    {
+        throw std::runtime_error("Initializing deflate failed");
+    }
+
+    ret = deflate(&zs, Z_FINISH);
+
+    if (ret != Z_STREAM_END)
+    {
+        std::cout << "Deflate failed for loading animtree file \"" << assetName << "\"" << std::endl;
```

Comment:
please consider using std::cerr for printing error messages (here and elsewhere in the pr)

Reply:
Fixed.